merlin/models/tf/core/transformations.py

```diff
@@ -40,6 +40,104 @@
 COUNT = utils.COUNT
 
 
+@tf.keras.utils.register_keras_serializable(package="merlin.models")
+class FeaturesTensorTypeConversion(TabularBlock):
+    """Base class to convert the tensor type of features provided in the schema
+
+    Parameters
+        ----------
+        schema : Optional[Schema], optional
+            The schema with the columns that will be transformed, by default None
+    """
+
+    def __init__(self, schema: Optional[Schema] = None, **kwargs):
+        super().__init__(**kwargs)
+        self.column_names = None
+        if schema is not None:
+            self.column_names = schema.column_names
+
+    def call(self, inputs: TabularData, **kwargs) -> TabularData:
+        outputs = {}
+        for name, val in inputs.items():
+            outputs[name] = val
+
+            if self.column_names is not None and name not in self.column_names:
+                continue
+
+            if isinstance(val, tuple):
+                val = list_col_to_ragged(val)
+            if isinstance(val, tf.RaggedTensor):
+                outputs[name] = val.to_sparse()
+            elif isinstance(val, tf.Tensor):
+                outputs[name] = tf.sparse.from_dense(val)
+
+        return outputs
+
+    def compute_output_shape(self, input_shapes):
+        output_shapes = {}
+        for k, v in input_shapes.items():
+            # If it is a list/sparse feature (in tuple representation), uses the offset as shape
+            if isinstance(v, tuple) and isinstance(v[1], tf.TensorShape):
+                output_shapes[k] = tf.TensorShape([v[1][0], None])
+            else:
+                output_shapes[k] = v
+
+        return output_shapes
+
+    def compute_call_output_shape(self, input_shapes):
+        return self.compute_output_shape(input_shapes)
+
+
+@Block.registry.register("to_sparse")
+@tf.keras.utils.register_keras_serializable(package="merlin.models")
+class ToSparseFeatures(FeaturesTensorTypeConversion):
```

Comment:
This naming confuses me. In the moving transforms PR, I am proposing to rename our current layers to `ListToSparse` & `ListToRagged` ([source](https://github.com/NVIDIA-Merlin/models/blob/de8faf947cc7399b547067903439225244180b4d/merlin/models/tf/transforms/tensor.py#L33)).

I think it would make more sense to call this one `ToSparse` and make it take in an optional schema in the constructor. If a schema is provided, only the features in the schema will be converted.

Reply:
Sure @marcromeyn. Renamed the new transformers to `ToSparse` and `ToDense`